fs/rails-base-graphql-api
# Add refresh token to jwt implementation

### Summary

Add refresh token to jwt implementation

### How it works

Sign In 
![image](https://user-images.githubusercontent.com/46126143/83244746-6805a700-a1a8-11ea-8272-92a4cdad30ab.png)

UpdateToken
![image](https://user-images.githubusercontent.com/46126143/83244866-a00cea00-a1a8-11ea-82be-23ca5ac9c443.png)


### Test plan

List of steps to manually test introduced functionality:

* Open Postman or other testing tools
* Make request using schema:
```
mutation {
        signin (
          email: "1@example.com",
          password: "12345678"
        ) {
          me {
            id
            email
          }
          refreshToken
          accessToken
        }
      }
```
* You should get an responce with `refresh_token`.

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Add this ENV variables:

```
AUTH_SECRET_TOKEN
```

### References
* [GitHub Issue #19](https://github.com/fs/rails-base-graphql-api/issues/19)
* [GitHub Issue #20](https://github.com/fs/rails-base-graphql-api/issues/20)
